dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/DocumentControllerTest.kt

```diff
@@ -91,13 +95,63 @@ class DocumentControllerTest {
         assertEquals(HttpStatus.NOT_FOUND.value(), exception.statusCode)
     }
 
+    @Test
+    fun `test that non admin users can upload documents only when they are CompanyOwner and not otherwise`() {
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        val testCompanyIdString = apiAccessor.uploadOneCompanyWithRandomIdentifier().actualStoredCompany.companyId
+        val testCompanyId = UUID.fromString(testCompanyIdString)
+        val dataReaderId = UUID.fromString(READER_USER_ID)
+
+        ensureUserIsNotOwnerOfAnyCompany(dataReaderId)
+
+        apiAccessor.companyRolesControllerApi.assignCompanyRole(
+            CompanyRole.CompanyOwner,
+            testCompanyId,
+            dataReaderId,
+        )
+
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+        assertDoesNotThrow { uploadDocument(pdfDocument, TechnicalUser.Reader) }
+
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        apiAccessor.companyRolesControllerApi.removeCompanyRole(
+            CompanyRole.CompanyOwner,
+            testCompanyId,
+            dataReaderId,
+        )
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+
+        val clientException = assertThrows<ClientException> {
+            uploadDocument(pdfDocument, TechnicalUser.Reader)
+        }
+        assertEquals("Client error : 403 ", clientException.message)
+    }
+
+    private fun ensureUserIsNotOwnerOfAnyCompany(userId: UUID) {
```

Comment:
Ich schlage vor dass wir die Funktion leicht umbennen, da sie zunächst den Eindruck erweckt, man würde nur eine Validierung durchführen.
In Wahrheit manipuliert man (potenziell) tatsächlich die DB, indem man role-assignments removet.

Vorschlag von mir wäre z.B. `private fun removePotentialCompanyOwnerRolesFromUser`
oder nur `private fun removeCompanyOwnerRolesFromUser`

Reply:
Oder noch besser `private fun removeAllCompanyOwnershipsFromUser`

Reply:
Done, warum man das macht ist ja auch klar ohne es explizit im Namen haben zu müssen